lib/model/model.go

```diff
@@ -2527,9 +2552,19 @@ func (m *model) CommitConfiguration(from, to config.Configuration) bool {
 	m.fmut.Lock()
 	for deviceID := range fromDevices {
 		delete(m.deviceStatRefs, deviceID)
+		// Forget all pending folders for removed devices
+		keepPendingFoldersFor[deviceID] = false //not really needed
 	}
 	m.fmut.Unlock()
 
+	// Forget pending devices that are now ignored
+	for _, ignDevice := range to.IgnoredDevices {
```

Comment:
ignored devices aren't going to the db?

Reply:
No, not intended.  They are deliberate user choices and I think it's useful they can be added in an XML config file manually or with any other XML capable tool without Syncthing running.